### DISCUSSION TITLE
Ionization Rate

### DISCUSSION THREAD

**Discussion by budjensen** [BLAST-WarpX/warpx#4936] [Ideas]
Is there a standard way to collect data on ionization rate versus position?

I've considered doing this in two ways:
1. Add a particle attribute which details the time and position of the particle creation event. I can then add a python function to periodically read these arrays and record the data.
2. Add callback functions in python and inspect newly created particles. I think this is less feasible given that there are no "pre-collisions" or "post-collisions" callback locations.

Would you have a recommendation of how to move forward on this?

Thank you!

**Comment by RemiLehe** [comment]
Thanks for the question.

In this WarpX example, for each electron resulting from an ionization event, we store the position at which the electron was created (as `orig_z`, you can then see this attribute in the openPMD output):
https://github.com/ECP-WarpX/WarpX/blob/development/Examples/Tests/ionization/inputs_2d_rt#L39

You could in principle do the same thing with the time at which the electrons were created:
```
electrons.addRealAttributes = orig_z orig_t
electrons.attribute.orig_z(x,y,z,ux,uy,uz,t) = z
electrons.attribute.orig_t(x,y,z,ux,uy,uz,t) = t
```

Does this answer your question?

**Reply by budjensen** [reply]
Thanks @RemiLehe, this is exactly what I'm looking for. Is there a way to set up this same initialization within a PICMI script?

**Reply by RemiLehe** [reply]
There isn't currently a way to do this in PICMI, but we might add this capability soon.

**Reply by roelof-groenewald** [reply]
The requested capabilities are being added in #4972.

**Reply by budjensen** [reply]
Thank you!
